test/integration/scratch-tests.js

```diff
@@ -0,0 +1,113 @@
+/* global vm, Promise */
+const {Chromeless} = require('chromeless');
+const test = require('tap').test;
+const path = require('path');
+const fs = require('fs');
+const chromeless = new Chromeless();
+
+const indexHTML = path.resolve(__dirname, 'index.html');
+const testDir = (...args) => path.resolve(__dirname, 'scratch-tests', ...args);
+
+const testFile = file => test(file, async t => {
+    // start each test by going to the index.html, and loading the scratch file
+    const says = await chromeless.goto(`file://${indexHTML}`)
+        .setFileInput('#file', testDir(file))
+        // the index.html handler for file input will add a #loaded element when it
+        // finishes.
+        .wait('#loaded')
+        .evaluate(() => {
+            // This function is run INSIDE the integration chrome browser via some
+            // injection and .toString() magic.  We can return some "simple data"
+            // back across as a promise, so we will just log all the says that happen
+            // for parsing after.
+
+            // this becomes the `says` in the outer scope
+            const messages = [];
+            const TIMEOUT = 5000;
+
+            vm.runtime.on('SAY', (_, __, message) => {
+                messages.push(message);
+            });
+
+            vm.greenFlag();
+            const startTime = Date.now();
+
+            return Promise.resolve()
+                .then(async () => {
+                    // waiting for all threads to complete, then we return
+                    while (vm.runtime.threads.length > 0) {
+                        if ((Date.now() - startTime) >= TIMEOUT) {
+                            messages.push(`fail Threads still running after ${TIMEOUT}ms`);
+                            break;
+                        }
+
+                        await new Promise(resolve => setTimeout(resolve, 50));
+                    }
+
+                    return messages;
+                });
+        });
+
+        // Map string messages to tap reporting methods. This will be used
+        // with events from scratch's runtime emitted on block instructions.
```

Comment:
Looks like these comments should be un-indented one notch

Reply:
fixed